### PR TITLE
Fix(ingest/bigquery): fix extracting comments from complex types

### PR DESCRIPTION
**Pull request by maaaikoool**
Closes https://github.com/datahub-project/datahub/issues/8763

Fixes a conditional to extract the field description when the field is part of a record. `SchemaField`s are lower case, while `BigQueryColumn`s have the original column name.

## Checklist

- [X ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [X] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Review by lusob — APPROVED** [review]

File: metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery.py [hunk 1]
```diff
@@ -1043,7 +1043,7 @@ def gen_schema_fields(self, columns: List[BigqueryColumn]) -> List[SchemaField]:
                         # Remove all the [version=2.0].[type=struct]. tags to get the field path
                         if (
                             re.sub(r"\[.*?\]\.", "", field.fieldPath, 0, re.MULTILINE)
```
Comment (mayurinehate) [inline metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery.py:1045]:
```suggestion
                            re.sub(r"\[.*?\]\.", "", field.fieldPath.lower(), 0, re.MULTILINE)
```

**Review by mayurinehate — CHANGES_REQUESTED** [review]
Tested with table having a complex type with uppercase leaf node and able to reproduce the issue.

If the root node of complex type is also in uppercase, the suggested change is required.
Otherwise looks good.

**Review by mayurinehate — APPROVED** [review]
LGTM

**Review by mayurinehate — CHANGES_REQUESTED** [review]
@maaaikoool can you please fix the lint ? https://github.com/datahub-project/datahub/actions/runs/6640885066/job/18042206118?pr=8950

**Review by mayurinehate — APPROVED** [review]

**Review by hsheth2 — APPROVED** [review]

**Pull request merged**